src/scripts/fetchStatsFromGA.js

```diff
@@ -0,0 +1,437 @@
+// eslint-disable no-console
+// TODO: consider the edge case when the cron job runs at midnight
+//       the first cron job of the day should also update the value for yesterday
+
+import 'dotenv/config';
+import client from 'util/client';
+import { google } from 'googleapis';
+import yargs from 'yargs';
+
+const analyticsreporting = google.analyticsreporting('v4');
+
+const ONEDAY = 1000 * 60 * 60 * 24;
+const maxDuration = 30;
+
+const pageSize = process.env.GA_PAGE_SIZE || '10000';
+const webViewId = process.env.GA_WEB_VIEW_ID;
+const lineViewId = process.env.GA_LINE_VIEW_ID;
+
+const idExtractor = /\/([^?/]+).*/;
+const dateFormat = /^[0-9]{4}-[0-9]{2}-[0-9]{2}$/;
+const toTitleCase = str =>
+  str.charAt(0).toUpperCase() + str.substr(1).toLowerCase();
+const formatDate = date =>
+  `${date.substr(0, 4)}-${date.substr(4, 2)}-${date.substr(6, 2)}`;
+
+const docTypes = {
+  ARTICLE: 'article',
+  REPLY: 'reply',
+};
+
+const allDocTypes = [docTypes.ARTICLE, docTypes.REPLY];
+
+const statsSources = {
+  WEB: {
+    filtersExpression: docType => `ga:pagePathLevel1==/${docType}/`,
+    name: 'WEB',
+    primaryDimension: (isCronjob = true) =>
+      isCronjob ? 'ga:contentGroup1' : 'ga:pagePathLevel2',
+    primaryMetric: 'ga:pageviews',
+    viewId: webViewId,
+  },
+  LINE: {
+    filtersExpression: docType =>
+      `ga:eventCategory==${toTitleCase(docType)};ga:eventAction==Selected`,
+    name: 'LINE',
+    primaryDimension: () => 'ga:eventLabel',
+    primaryMetric: 'ga:hits',
+    viewId: lineViewId,
+  },
+};
+
+const allSourceTypes = [statsSources.WEB.name, statsSources.LINE.name];
+
+const upsertScript = `
+  // since web stats and line stats are fetched seperately, need to update each
+  // stats field seperately so existing values won't be overwritten
+  if (ctx._source.size() == 0 || ctx._source.stats.size() == 0) {
+    ctx._source.stats = params.stats;
+  } else {
+    if (params.stats.webUser != null) {
```

Comment:
I am wondering if [`putAll()`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html#putAll(java.util.Map)) is useful here. 

I also like the current implementation, which explicitly lists all changed fields, this makes modifications crystal clear 💎 .